packages/react-native-codegen/src/parsers/typescript/utils.js

```diff
@@ -50,7 +50,7 @@ export type ASTNode = Object;
 
 const invariant = require('invariant');
 
-type TypeAliasResolutionStatus =
+export type TypeAliasResolutionStatus =
```

Comment:
I think we should share this type between `flow` and `typescript` folders. Unfortunately, we can't share all the logic.

Should i move this type only in `packages/react-native-codegen/src/parsers/utils.js`? It is the closest common `utils` file between flow and typescript.

Or, should it be another task @cipolleschi?

Reply:
Yes, please! 😄 
If you can move this type up to the `parsers/utils.js` and reuse it, it would be amazing!

> Unfortunately, we can't share all the logic.
What do you mean with this? The type definition seems identical to me. Which logic you feel like we can't share? 🤔

Reply:
> What do you mean with this? The type definition seems identical to me. Which logic you feel like we can't share? 🤔

I mean we can share `TypeAliasResolutionStatus` which is identical but not `resolveTypeAnnotation` which uses this type. 

If it had been possible, I would have moved everything in `react-native-codegen/src/parsers/utils.js` 😄

Reply:
Ah yeah... `resolveTypeAnnotation` is another beast. One of the goals of the umbrella task is to make those two methods as close as possible with incremental changes so that, at some point, we may be able to put them together! 

It is expected that it is not the time... yet! :D

Reply:
yes, let's refactor step by step 😁

I moved the type!